firebase-firestore-mixin.html

```diff
@@ -0,0 +1,214 @@
+   * the templatized properties change.
+   *
+   * PolymerFirestore bindings are **intentionally read-only**. Automatic
+   * three-way binding (i.e. syncing changes from the element back up to
+   * the database) are great for toy apps but largely an antipattern.
+   *
+   * In addition to loading data into the specified property, PolymerFirestore
+   * also makes additional convenience properties:
+   *
+   * * `<propname>Ref`: a Firestore reference to the doc/collection
+   * * `<propname>Ready`: will be true when all path segments are present and data has been fetched at least once
+   *
+   * ### Querying
+   *
+   * PolymerFire supports querying by supplying a builder function to the
+   * `query` option. The function will be bound to the element and called with
+   * the ref and element instance. For example:
+   *
+   * ```js
+   * {
+   *   uid: String,
+   *   label: String,
+   *   category: String,
+   *   notes: {
+   *     type: Array,
+   *     collection: 'users/{uid}/notes',
+   *     query: (q, el) => {
+   *       q = q.orderBy('date', 'desc').limit(100)
+   *       if (el.color) { q.where('color','==', el.color) }
+   *       if (el.category) { q.where('category', '==', el.category) }
+   *       return q;
+   *     },
+   *     observes: ['color', 'category']
+   *   }
+   * }
+   * ```
+   *
+   * In the  above example, a rich query is further filtered down by other
+   * properties on the element. Remember to declare each query-affecting
+   * property in your `observes` option.
+   *
+   * ### Options
+   *
+   * * **doc:** *string*, the full (optionally templatized) path to a document
+   * * **collection:** *string*, the full (optionally templatized) path to a collection
+   * * **live:** *boolean*, whether or not to continue updating the property as data changes in the database
+   * * **query:** *(ref: Reference, el: Element)* a query builder function that takes the computed ref and the element instance
+   * * **observes:** a list of properties which, if changed, should trigger a rebuild of the query
+   *
+   * @polymer
+   * @mixinFunction Polymer.FirestoreMixin
+   */
+  Polymer.FirestoreMixin = parent => {
+    return class extends parent {
+      constructor() {
+        super();
+        this._firestoreProps = {};
+        this._firestoreListeners = {};
+        this.db = this.constructor.db || firebase.firestore();
+      }
+
+      connectedCallback() {
+        const props = collect(this.constructor, 'properties');
+        for (let name in props) {
+          if (props[name].doc) {
+            this._firestoreBind('doc', props[name].doc, name, props[name].live, props[name].observes);
+          } else if (props[name].collection) {
+            this._firestoreBind('collection', props[name].collection, name, props[name].live, props[name].observes);
+          }
+        }
+        super.connectedCallback();
+      }
+
+      _firestoreBind(type, path, name, live = false, observes = []) {
```

Comment:
Isn't here the query missing.
Shouldn't it, so it's available in this line 194.
I looked through the code because I was wondering why queries are never called and realised that query is never assigned to the config.

Is this a correct discovery?
Would love to see this fixed if so.
:)

Reply:
https://github.com/firebase/polymerfire/pull/303/files

Reply:
@merlinnot Is there a version which fixes this issue and can be used for internal tools?

Reply:
@IchordeDionysos Not really. I've made quite a few PRs to fix various issues in a Firestore mixin, but repo owners are really unresponsive :cry: Take a look at https://github.com/firebase/polymerfire/issues/274.